File: app/src/main/java/com/example/helpwheel/ui/fuel_management/new_ride_fragments/SpendFuelNewRideFragment.java
package com.example.helpwheel.ui.fuel_management.new_ride_fragments;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.helpwheel.R;
import com.example.helpwheel.databinding.FragmentSpendFuelNewRideBinding;
import com.example.helpwheel.utils.Constants;

public class SpendFuelNewRideFragment extends Fragment {
    FragmentSpendFuelNewRideBinding binding;
    Constants constants;
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentSpendFuelNewRideBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        SharedPreferences fuelStats = requireContext().getSharedPreferences(constants.APP_PREFERENCES, requireContext().MODE_PRIVATE);
        binding.spendFuelNewRide.setText(String.format("%s %s", fuelStats.getFloat(constants.APP_NEW_RIDE_WILL_BE_USED_FUEL, 0.0f), getString(R.string.litres_symbol)));
    }
}

File: app/src/main/java/com/example/helpwheel/ui/fuel_management/new_ride_fragments/SpendFuelNewRideFragment.kt
package com.example.helpwheel.ui.fuel_management.new_ride_fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.example.helpwheel.databinding.FragmentSpendFuelNewRideBinding
import com.example.helpwheel.ui.fuel_management.TripViewModel

class SpendFuelNewRideFragment : Fragment() {
    lateinit var binding: FragmentSpendFuelNewRideBinding
    private lateinit var tripViewModel: TripViewModel

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentSpendFuelNewRideBinding.inflate(inflater, container, false)
        tripViewModel = ViewModelProvider(requireActivity())[TripViewModel::class.java]
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        tripViewModel.getSpendFuelNewRide()
            .observe(viewLifecycleOwner) { binding.spendFuelNewRide.text = it }
    }

}